pkg/scheduler/flavorassigner/flavorassigner_test.go

```diff
@@ -2409,7 +2409,7 @@ func TestAssignFlavors(t *testing.T) {
 				t.Errorf("e.assignFlavors(_).RepresentativeMode()=%s, want %s", repMode, tc.wantRepMode)
 			}
 
-			if diff := cmp.Diff(tc.wantAssignment, assignment, cmpopts.IgnoreUnexported(Assignment{}, FlavorAssignment{}), cmpopts.IgnoreFields(Assignment{}, "LastState"), cmpopts.IgnoreFields(FlavorAssignment{}, "TriedFlavorIdx")); diff != "" {
+			if diff := cmp.Diff(tc.wantAssignment, assignment, cmpopts.IgnoreUnexported(Assignment{}, FlavorAssignment{}), cmpopts.IgnoreFields(Assignment{}, "LastState")); diff != "" {
```

Comment:
+1, would any of the pre-existing test cases fail after the change? If not please add a dedicated one.

Reply:
yes, "each podset requires preemption on a different flavor" will fail after the change.
```
Unexpected assignment (-want,+got):
          flavorassigner.Assignment{
          	PodSets: []flavorassigner.PodSetAssignment{
          		{
          			Name: "launcher",
          			Flavors: flavorassigner.ResourceAssignment{
          				s"cpu": &{
          					Name:           "one",
          					Mode:           s"Preempt",
        - 					TriedFlavorIdx: -1,
        + 					TriedFlavorIdx: 0,
          					... // 1 ignored field
          				},
          			},
          			Status:   &{reasons: {...}},
          			Requests: {s"cpu": {i: {...}, Format: "DecimalSI"}},
          			Count:    1,
          		},
          		{Name: "workers", Flavors: {s"cpu": &{Name: "tainted", Mode: s"Preempt", TriedFlavorIdx: -1}}, Status: &{reasons: {...}}, Requests: {s"cpu": {i: {...}, Format: "DecimalSI"}}, ...},
          	},
          	Borrowing: false,
          	... // 2 ignored and 1 identical fields
          }
```

For launcher, flavor 1 is tainted, so in the past triedFlavor idx would be 0 and now triedFlavor idx would be -1.

Reply:
ack